src/core/ColorPicker/ColorPicker.tsx

```diff
@@ -0,0 +1,546 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import '@itwin/itwinui-css/css/color-picker.css';
+import {
+  useTheme,
+  CommonProps,
+  getBoundedValue,
+  getWindow,
+  useEventListener,
+} from '../utils';
+import cx from 'classnames';
+import tinyColor from 'tinycolor2';
+
+export type Color = {
+  hsl?: HslColor;
+  rgb?: RgbColor;
+  hex?: HexColor;
+  hsv?: HsvColor;
+};
+
+interface HslColor {
+  h: number;
+  s: number;
+  l: number;
+  displayString?: string;
+}
+interface RgbColor {
+  r: number;
+  g: number;
+  b: number;
+  displayString?: string;
+}
+interface HexColor {
+  hex: string;
+}
+interface HsvColor {
+  h: number;
+  s: number;
+  v: number;
+  displayString?: string;
+}
+
+const getVerticalPercentageOfRectangle = (rect: DOMRect, pointer: number) => {
+  const position = getBoundedValue(pointer, rect.top, rect.bottom);
+  return ((position - rect.top) / rect.height) * 100;
+};
+const getHorizontalPercentageOfRectangle = (rect: DOMRect, pointer: number) => {
+  const position = getBoundedValue(pointer, rect.left, rect.right);
+  return ((position - rect.left) / rect.width) * 100;
+};
+
+// Converts color value provided and fills in all hsv, hsl, rgb, hex, and displayString values
+export const fillColor = (color: Color | undefined) => {
+  let c;
+
+  if (color == undefined) {
+    c = tinyColor('#ff0000');
+  } else if (color.hsv != undefined) {
+    c = tinyColor(color.hsv);
+  } else if (color.hsl != undefined) {
+    c = tinyColor(color.hsl);
+  } else if (color.rgb != undefined) {
+    c = tinyColor(color.rgb);
+  } else if (color.hex != undefined) {
+    c = tinyColor(color.hex.hex);
+  } else {
+    c = tinyColor('#ff0000');
+  }
+
+  const hsv = c.toHsv();
+  const hsl = c.toHsl();
+  const rgb = c.toRgb();
+
+  return {
+    hsv: { h: hsv.h, s: hsv.s, v: hsv.v, displayString: c.toHsvString() },
+    hsl: { h: hsl.h, s: hsl.s, l: hsl.l, displayString: c.toHslString() },
+    rgb: { r: rgb.r, g: rgb.g, b: rgb.b, displayString: c.toRgbString() },
+    hex: { hex: c.toHex() },
+  };
+};
+
+export type ColorPickerProps = {
```

Comment:
Would need to think about the naming, but I feel we could still have colors as children and get rid of `onAdd`. Instead, we could support ability to have that (passing add button together with `ColorSwatch` components as children). That would be more flexible for what ever people might want to have there. Do you guys think it would make sense? @bsteinbk @saaaaaally @mayank99

Reply:
@saaaaaally - That sounds OK to me. I plan to get back to making these updates to ColorValue later today or early Monday.